llvm/lib/Transforms/Vectorize/VPlan.h

```diff
@@ -1769,6 +1772,47 @@ struct LLVM_ABI_FOR_TEST VPWidenSelectRecipe : public VPRecipeWithIRFlags,
   }
 };
 
+/// A recipe for selecting whole vector values based on a scalar condition.
```

Comment:
do we need a dedicated recipe for this? If we have a scalar condition, broadcasting should be fine, although more vebose than needed? 

We should be able to handle single-scalar conditions as well, which may be enough? https://github.com/llvm/llvm-project/pull/165506#issuecomment-3459535511